lti_emailer/views.py

```diff
@@ -56,4 +59,11 @@ def lti_launch(request):
         "lti_emailer launched with params: %s",
         json.dumps(request.POST.dict(), indent=4)
     )
-    return redirect('mailing_list:index', request.POST['resource_link_id'])
+
+    view = 'mailing_list'
+    if has_lti_role(request, const.LEARNER):
```

Comment:
Don't you get all roles in Canvas with every LTI launch, or is it just your roles for the course? I think that in the ENV variable Rails provides in the global Javascript context the roles are the set of all roles you have for all courses/accounts, but maybe this is different. If it is the sum total of your roles, this might not be accurate, but if it is only course-level roles, that makes sense.